docs/create-webhook.yaml

```diff
@@ -0,0 +1,159 @@
+    - name: GithubOwner
+      description: The github owner name
+      default: "owner"
+    - name: GithubRepo
+      description: The github repo name 
+      default: "repo"
+    - name: GithubSecretName
+      description: The secret name for github access token 
+      default: "githubsecret"
+    - name: GithubUrl
+      description: The url of git hub 
+      default: "github.com"
+    - name: EventListenerName
+      description: The event listener name 
+      default: "eventlistener"
+    - name: TriggerBinding
+      description: The Trigger binding 
+      default: "triggerbinding"
+    - name: TriggerTemplate
+      description: The Trigger Template 
+      default: "triggertemplate"
+    - name: TriggerServiceAccount
+      description: The Trigger service account 
+      default: "default"
+  steps:
+  - name: generate-certificate
+    image: lachlanevenson/k8s-kubectl:latest
+    command:
+    - sh
+    args:
+    - -ce
+    - |
+      set -e
+      cat <<EOF | sh
+      #!/bin/sh
+      if [ $(inputs.params.CreateCertificate) = "false" ]
+      then
+        exit 0
+      fi
+      apk update
+      apk add openssl
+      mkdir /root/ingress
+      openssl genrsa -des3 -out /root/ingress/key.pem -passout pass:$(inputs.params.CertificateKeyPassphrase) 2048
+      openssl req -x509 -new -nodes -key /root/ingress/key.pem -sha256 -days 1825 -out /root/ingress/certificate.pem -passin pass:$(inputs.params.CertificateKeyPassphrase) -subj /CN=$(inputs.params.ExternalUrl)
+      openssl rsa -in /root/ingress/key.pem -out /root/ingress/key.pem -passin pass:$(inputs.params.CertificateKeyPassphrase)
+      kubectl create secret tls $(inputs.params.CertificateSecretName) --cert=/root/ingress/certificate.pem --key=/root/ingress/key.pem -n tekton-pipelines
+      EOF
+  - name: create-ingress
+    image: lachlanevenson/k8s-kubectl:latest
+    command:
+    - sh
+    args:
+    - -ce
+    - |
+      set -e
+      if [ $(inputs.params.CreateIngress) = "false" ]
+      then
+        exit 0
+      fi
+      cat <<EOF | kubectl apply -f -
+      apiVersion: extensions/v1beta1
+      kind: Ingress
+      metadata:
+        name: $(inputs.params.EventListenerName)
+      spec:
+        tls:
+        - hosts:
+          - $(inputs.params.ExternalUrl)
+          secretName: $(inputs.params.CertificateSecretName)
+        rules:
+        - host: $(inputs.params.ExternalUrl)
+          http:
+            paths:
+            - backend:
+                serviceName: $(inputs.params.EventListenerName)
+                servicePort: 8082
+      EOF
+  - name: create-webhook
+    image: alpine:latest
+    command:
+    - sh
+    args:
+    - -ce
+    - |
+      set -e
+      if [ $(inputs.params.CreateWebhook) = "false" ]
+      then
+        exit 0
+      fi
+      apk update
+      apk add curl
+      echo "Create Webhook"
+      if [ $(inputs.params.GithubUrl) = "github.com" ]
+      then
```

Comment:
Don't we need secret also for securing endpoint?

Reply:
Yes, eventually.  The handling of the github secret is not in the event listener yet.  When it is implemented, the secret should be added.